ifttt/watches/gcloud_datastore_aggregation.py

```diff
@@ -19,17 +22,27 @@ def __repr__(self):
         return "AggregatedDatastoreWatch '{}'".format(self.name)
 
     def collect_activations(self):
-        collector = list()
+        collector = collections.defaultdict(list)
 
         query = self.client.query(kind=self.kind)
         for result in query.fetch():
             value = result.get(self.field)
             if not value:
                 continue  # TODO: aggregates across nulls
 
-            collector.append(value)
+            context = None
+            if self.context_field:
+                context = result.get(self.context_field)
+                context = context or None  # TODO: consider handling nulls vs ''
 
-        prev = self.cache[self.cache_key].get(self.field, 0)  # TODO: aggregates across nulls
-        curr = self.aggregate_fn(collector)
-        if self.if_fn(self.cache_key, prev, curr):
-            yield self.cache_key, curr
+            collector[context].append(value)
```

Comment:
If no context field exists, this will throw a `KeyError` and ham the whole collection activation. Is that expected (i.e. no except)?

Reply:
Am I missing something?

```
kevin@atomicbox ~ » python3
>>> import collections
>>> collector = collections.defaultdict(list)
>>> collector[None].append('aardvark')
>>> collector
defaultdict(<class 'list'>, {None: ['aardvark']})
```

Reply:
Good lord, I didn't know that was possible.